### PR TITLE
Add a user guide and architectural description with rationale

### PR DESCRIPTION
**Pull request by jypma**

### CODE REVIEW

File: guide.MD [hunk 1]
```diff
@@ -0,0 +1,113 @@
...
+to add a timestamp to events and instruct clients to use that for ordering.   
+
+This use guide assumes basic familiarity with [akka's documentation](http://doc.akka.io/docs/akka/current/java.html), 
+especially the akka persistence sections. All of `ts-reaktive` assumes deployment into a clustered akka setup.
+
+When to use event sourcing
+--------------------------
+Event sourcing is a great choice if you expect to have use cases in the following categories:
+
+- Having historic data affect current actions
```
Comment (alar17) [inline guide.MD]: affects

File: guide.MD [hunk 2]
```diff
@@ -0,0 +1,113 @@
...
+So you should pick your aggregate sizes big enough to allow important business logic to be implemented (for which 
+"eventual consistency" isn't good enough). At the same time, the aggregates need to be small enough so that your
+millions of concurrently connected users still represent millions of aggregates, so they can be distributed across
+a cluster.
+
+Writing an event-sourced actor
+------------------------------
+Within `ts-reaktive`, an event-sourced actor defines concrete type for the _commands_ and _events_ it deals with. This
+is contrary to plain akka, where both of these are just `Object`. In addition, an event-sourced actor externalizes
+its complete in-memory state into an external type as well. That has the advantage that that "state" class can be
```
Comment (alar17) [inline guide.MD]: So many "that"s. Could you rephrase it?

**Comment by jypma** [comment]
thanks a lot! updated and merging.
